### PR TITLE
Adding class-string return types

### PR DESCRIPTION
**Pull request by msmakouz**

### CODE REVIEW

File: src/Definition/Entity.php [hunk 1]
```diff
@@ -125,6 +134,9 @@ public function setScope(?string $scope): self
         return $this;
     }
 
+    /**
+     * @return class-string|null
```
Comment (roxblnfk) [inline src/Definition/Entity.php]: Shouldn't it be some `class-string<ScopeInterface>|null`?

File: src/Definition/Entity.php [hunk 2]
```diff
@@ -137,6 +149,9 @@ public function setRepository(?string $repository): self
...
```
Comment (roxblnfk) [inline src/Definition/Entity.php]: Shouldn't it be some `class-string<RepositoryInterface>|null`?

File: src/Definition/Entity.php [hunk 3]
```diff
@@ -113,6 +119,9 @@ public function setSource(?string $source): self
...
```
Comment (roxblnfk) [inline src/Definition/Entity.php]: Shouldn't it be some `class-string<SourceInterface>|null`?

File: src/Definition/Entity.php [hunk 4]
```diff
@@ -89,6 +89,9 @@ public function setClass(string $class): self
...
```
Comment (roxblnfk) [inline src/Definition/Entity.php]: You can add class template like TEntity and use it here and with other interfaces like
`class-string<RepositoryInterface<TEntity>>|null`?

File: src/Definition/Entity.php (hunk 1 above)
Comment (msmakouz) [reply src/Definition/Entity.php]: Added ```class-string<ScopeInterface>|null```

File: src/Definition/Entity.php (hunk 2 above)
Comment (msmakouz) [reply src/Definition/Entity.php]: Added ```class-string<RepositoryInterface>|null```

File: src/Definition/Entity.php (hunk 3 above)
Comment (msmakouz) [reply src/Definition/Entity.php]: Added ```class-string<SourceInterface>|null```

File: src/Definition/Entity.php (hunk 4 above)
Comment (msmakouz) [reply src/Definition/Entity.php]: Added template

File: src/Definition/Entity.php [hunk 5]
```diff
@@ -105,7 +111,7 @@ public function setMapper(?string $mapper): self
     }
 
     /**
-     * @return class-string|null
+     * @return class-string<MapperInterface>|null
```
Comment (roxblnfk) [inline src/Definition/Entity.php:153]:
```suggestion
     * @return class-string<MapperInterface<TEntity>>|null
```

File: src/Definition/Entity.php [hunk 6]
```diff
@@ -150,7 +156,7 @@ public function setRepository(?string $repository): self
     }
 
     /**
-     * @return class-string|null
+     * @return class-string<RepositoryInterface>|null
```
Comment (roxblnfk) [inline src/Definition/Entity.php]:
```suggestion
     * @return class-string<RepositoryInterface<TEntity>>|null
```

File: src/Definition/Entity.php (hunk 5 above)
Comment (roxblnfk) [reply src/Definition/Entity.php:153]: MapperInterface has no template

**Review by roxblnfk — APPROVED** [review]

**Pull request merged**